### PR TITLE
docs: add numeric ranges and other bits

### PR DESCRIPTION
**Pull request by JimGalasyn**
### Description 
Add numeric ranges and a few other bits of info.

### CODE REVIEW

File: docs/reference2/sql-syntax/data-types/compound.md [hunk 1]
```diff
@@ -21,8 +21,8 @@ ksqlDB supports fields that are arrays of another type. All of the elements
 in the array must be of the same type. The element type can be any valid
 SQL type.
 
-The elements of an array are zero-indexed and can be accessed by using
-the `[]` operator passing in the index. For example, `SOME_ARRAY[0]`
+The elements of an array are one-indexed and can be accessed by using
```
Comment (JimGalasyn) [inline docs/reference2/sql-syntax/data-types/compound.md:24]: I had to find the code to prove to myself this is true: https://github.com/confluentinc/ksql/blob/master/ksqldb-execution/src/main/java/io/confluent/ksql/execution/codegen/helpers/ArrayAccess.java.

File: docs/reference2/sql-syntax/data-types/numeric.md [hunk 2]
```diff
@@ -30,6 +28,18 @@ The `double` data type is an inexact, variable-precision numeric type. The term
 "inexact" means an approximate value is stored. Storing values outside of its
 bounds of capacity will result in an error.
 
+## Valid ranges
+
+Numeric data types have the same valid minimum and maximum values as their
+corresponding Java types. The following table shows valid ranges for the
+integer and floating-point types. 
+
+| type     | minimum value | maximum value
```
Comment (MichaelDrogalis) [inline docs/reference2/sql-syntax/data-types/numeric.md:37]: Can we move these to the top table to keep them in one place?

File: docs/reference2/sql-syntax/data-types/numeric.md [hunk 3]
```diff
@@ -30,6 +28,18 @@ The `double` data type is an inexact, variable-precision numeric type. The term
...
+|----------|---------------|--------------
+| `int`    | [Integer.MIN_VALUE](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Integer.html#MIN_VALUE) = -2<sup>31</sup> | [Integer.MAX_VALUE](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Integer.html#MAX_VALUE) = 2<sup>31</sup>-1
+| `bigint` | -2<sup>Integer.MAX_VALUE</sup> (exclusive) | 2<sup>Integer.MAX_VALUE</sup> (exclusive)
```
Comment (MichaelDrogalis) [inline docs/reference2/sql-syntax/data-types/numeric.md:40]: I don't believe `bigint` has a min/max range. It's only bounded by resources.

File: docs/reference2/sql-syntax/data-types/compound.md (hunk 1 above)
Comment (MichaelDrogalis) [reply docs/reference2/sql-syntax/data-types/compound.md:24]: Yes, `1` is right. I copied in some older docs here.

File: docs/reference2/sql-syntax/data-types/numeric.md (hunk 3 above)
Comment (blueedgenick) [reply docs/reference2/sql-syntax/data-types/numeric.md:40]: it maps to a Java Long internally - a signed 64-bit integer. So _almost_ unlimited, has a lot of digits!

Comment (MichaelDrogalis) [reply docs/reference2/sql-syntax/data-types/numeric.md:40]: Oh wow, today I learned. I thought we used `java.math.BigInt` under the hood. Looks like you're right:

```
ksql> insert into s1 (x) values (9223372036854775807);
ksql> select * from s1 emit changes;
+---------------------------------------------------------------------------------------------------------------------------+
|X                                                                                                                          |
+---------------------------------------------------------------------------------------------------------------------------+
|9223372036854775807                                                                                                        |
^CQuery terminated
ksql> insert into s1 (x) values (9223372036854775808);
Failed to prepare statement: line 1:29: Invalid numeric literal: 9223372036854775808
```

Thanks Nick!

File: docs/reference2/sql-syntax/data-types/numeric.md (hunk 2 above)
Comment (JimGalasyn) [reply docs/reference2/sql-syntax/data-types/numeric.md]: I moved this into the main table and tried different arrangements, but still it renders a bit awkwardly: 
image.
Columns get a bit cramped and line breaks happen in weird places. Probably okay for now, but we might want to revisit.

Comment (MichaelDrogalis) [reply docs/reference2/sql-syntax/data-types/numeric.md]: Np, we can figure out column widths before final merge.

**Pull request merged**